src/assessments/seo/textImagesAssessment.js

```diff
@@ -102,16 +148,64 @@ class TextImagesAssessment extends Assessment {
 			};
 		}
 
-		// Has alt-tag, but no keyword is set
-		if ( this.altProperties.withAlt > 0 ) {
+		// Image count ≥5, has alt-tags with too few keywords.
+		if ( this.hasTooFewMatches() ) {
 			return {
-				score: this._config.scores.withAlt,
-				resultText: i18n.dgettext(
-					"js-text-analysis",
-					"The images on this page contain alt attributes."
+				score: this._config.scores.withAltTooFewKeywordMatches,
+				resultText: i18n.sprintf(
+					/* Translators: %1$d expands to the number of images containing an alt attribute with the keyword,
+					 * %2$d expands to the total number of images. */
+					i18n.dgettext(
+						"js-text-analysis",
+						"Only in %1$d out of %2$d images on this page contain alt attributes with the focus keyword. " +
+						"Where appropriate, try to include the focus keyword in more alt attributes."
+					),
+					this.altProperties.withAltKeyword,
+					this.imageCount
+				),
+			};
+		}
+
+		/*
+		 * This check needs to be made before the check for too many matches because of the special rule for
+		 * exactly 5 matches.
+		 */
+		if ( this.hasGoodNumberOfMatches() ) {
+			return {
+				score: this._config.scores.withAltGoodNumberOfKeywordMatches,
+				resultText: i18n.sprintf(
+					/* Translators: %1$d expands to the number of images containing an alt attribute with the keyword,
+                     * %2$d expands to the total number of images. */
+					i18n.dngettext(
+						"js-text-analysis",
+						"The image on this page contains an alt attribute with the focus keyword.",
+						"%1$d out of %2$d images on this page contain alt attributes with the focus keyword.",
+						this.imageCount
+					),
+					this.altProperties.withAltKeyword,
+					this.imageCount
+				),
+			};
+		}
+
+		if ( this.hasTooManyMatches() ) {
+			return {
+				score: this._config.scores.withAltTooFewKeywordMatches,
+				resultText: i18n.sprintf(
+					/* Translators: %1$d expands to the number of images containing an alt attribute with the keyword,
+                     * %2$d expands to the total number of images. */
+					i18n.dgettext(
+						"js-text-analysis",
+						"%1$d out of %2$d images on this page contain alt attributes with the focus keyword. " +
+						"That's a bit much. Only include the focus keyword when it really fits the image."
+					),
+					this.altProperties.withAltKeyword,
+					this.imageCount
 				),
 			};
 		}
+
+		// Images, but no alt tags.
```

Comment:
This condition is triggered if there are 4 or less images with empty alt-tags.
<img width="330" alt="screen shot 2018-06-15 at 09 43 34" src="https://user-images.githubusercontent.com/36954173/41457273-6df90830-7083-11e8-8566-540c88bc0ace.png">
However, if there are 5 or more images with empty alt-tags, an earlier condition is triggered by `hasTooFewMatches()`, hence the feedback is different.
<img width="539" alt="screen shot 2018-06-15 at 09 43 05" src="https://user-images.githubusercontent.com/36954173/41457363-ba3869de-7083-11e8-8a3f-00db51b51f12.png">
In my eyes the latter looks really bad.

Please consider adding a check to `hasTooFewMatches` this.altProperties.withAltKeyword>0. Or suggest an alternative solution.

Reply:
Even when this `hasTooFewMatches()` is triggered correctly (e.g., for cases when there are >4 images and alt-tags are not empty but none of them has a keyword), the feedback string "Only (in - _hope you remove it!_) 0 out of 8 images on this page contain alt attributes with the focus keyword." sounds awkward. Could you please consider adding a separate string for those zero-cases? Something like "None of the 8 images on this page...".

Reply:
Ah yes, that wasn't working the way it was supposed to. I've added `this.altProperties.withAltKeyword>0` to the `hasTooFewMatches()` condition.

When there are >4 images and alt-tags are not empty but none of them has a keyword, the feedback is `The images on this page do not have alt attributes containing the focus keyword.`, so there should not be a problem with zero-cases.